server/api/sport/sport.controller.js

```diff
@@ -72,9 +72,10 @@ exports.destroy = function(req, res) {
 exports.populate = function(req, res) {
     Sport.find(req.params.id).then(function (sport) {
         if(!sport) { return res.send(404); }
-        parseCsv(sport)
+        return parseCsv(sport)
+    }).then(function() {
         return res.send(201, sport);
-    }, function(error){
+    }.catch(function(err) {
```

Comment:
My tendency is to only refactor code I'm actively touching unless it's actually broken, so there's no need to go out of your way to switch it around.